ember_debug/view-debug.js

```diff
@@ -4,54 +4,56 @@ import RenderTree from 'ember-debug/libs/render-tree';
 import ViewInspection from 'ember-debug/libs/view-inspection';
 import bound from 'ember-debug/utils/bound-method';
 
-import { readOnly } from 'ember-debug/utils/ember/object/computed';
-
-export default DebugPort.extend({
-  namespace: null,
-
-  adapter: readOnly('namespace.adapter'),
-  objectInspector: readOnly('namespace.objectInspector'),
-
-  portNamespace: 'view',
-
-  messages: {
-    getTree({ immediate }) {
-      this.sendTree(immediate);
-    },
-
-    showInspection({ id, pin }) {
-      this.viewInspection.show(id, pin);
-    },
-
-    hideInspection() {
-      this.viewInspection.hide();
-    },
-
-    inspectViews({ inspect }) {
-      if (inspect) {
-        this.startInspecting();
-      } else {
-        this.stopInspecting();
-      }
-    },
-
-    scrollIntoView({ id }) {
-      this.renderTree.scrollIntoView(id);
-    },
-
-    inspectElement({ id }) {
-      this.renderTree.inspectElement(id);
-    },
-
-    contextMenu() {
-      let { lastRightClicked } = this;
-      this.lastRightClicked = null;
-      this.inspectNearest(lastRightClicked);
-    },
-  },
+export default class extends DebugPort {
+  get adapter() {
+    return this.namespace?.adapter;
+  }
+  get objectInspector() {
+    return this.namespace?.objectInspector;
+  }
+
+  static {
+    this.prototype.portNamespace = 'view';
+
+    this.prototype.messages = {
+      getTree({ immediate }) {
+        this.sendTree(immediate);
+      },
+
+      showInspection({ id, pin }) {
+        this.viewInspection.show(id, pin);
+      },
+
+      hideInspection() {
+        this.viewInspection.hide();
+      },
+
+      inspectViews({ inspect }) {
+        if (inspect) {
+          this.startInspecting();
+        } else {
+          this.stopInspecting();
+        }
+      },
+
+      scrollIntoView({ id }) {
+        this.renderTree.scrollIntoView(id);
+      },
+
+      inspectElement({ id }) {
+        this.renderTree.inspectElement(id);
+      },
+
+      contextMenu() {
+        let { lastRightClicked } = this;
+        this.lastRightClicked = null;
+        this.inspectNearest(lastRightClicked);
+      },
+    };
+  }
 
   init() {
```

Comment:
Should this be `init` still or should we update it to `constructor`?

Reply:
i left init, because it gets overwritten in some parts in the tests. I didn't check which ones exactly